azure-functions/acbs-function/src/functions/activity-update-facility-guarantee.js

```diff
@@ -1,56 +1,43 @@
- * If the API request fails, it throws an error with details about the request and the error.
- * If the payload is not valid or does not contain all mandatory fields, it returns an object with the missing mandatory fields.
- * If any other error occurs, it logs the error and throws a new error.
+ * Handles the amendment of a facility guarantee record in the ACBS system.
+ *
+ * This function performs the following operations:
+ * 1. Validates the input payload.
+ * 2. Submits the amendment to the ACBS system.
+ * 3. Handles the response from the ACBS system and returns the result.
  *
- * @param {Object} payload - The payload containing the facilityIdentifier and acbsFacilityGuaranteeInput.
- * @param {string} payload.facilityIdentifier - The identifier of the facility.
- * @param {import('../../mappings/facility/facility-guarantee-amend').MappedFacilityGuaranteeAmendment} payload.acbsFacilityGuaranteeInput - The acbsFacilityGuaranteeInput object containing the guarantee details.
- * @returns {Object} - An object containing the status, timestamps of when the request was sent and received, the data sent, and the data received from the API.
- * @throws {Error} - Throws an error if the payload is invalid, if the API request to generate the guarantee ID fails, or if any other error occurs.
+ * @param {Object} payload - The input payload containing the facility ID and ACBS facility guarantee input.
+ * @param {string} payload.facilityId - The ID of the facility to be amended.
+ * @param {Object} payload.acbsFacilityGuaranteeInput - The ACBS facility guarantee input details.
+ * @returns {Object} - The result of the facility guarantee record amendment, including status, timestamps, and data sent/received.
+ * @throws {Error} - Throws an error if the input payload is invalid or if there is an error during the amendment process.
  */
 const handler = async (payload) => {
   try {
     if (!payload) {
       throw new Error('Invalid facility guarantee update payload');
     }
 
-    const { facilityIdentifier, acbsFacilityGuaranteeInput } = payload;
+    /**
+     * There are no mandatory fields for guarantee amendment payload
+     * `expirationDate` is only set for cover end date attrbiute.
+     * `guaranteedLimit` is only set for the value attribute.
+     */
 
-    // Call create guarantee API
+    const { facilityId, acbsFacilityGuaranteeInput } = payload;
     const submittedToACBS = getNowAsIsoString();
-    const { status, data } = await api.updateFacilityGuarantee(facilityIdentifier, acbsFacilityGuaranteeInput);
+    const { status, data } = await api.updateFacilityGuarantee(facilityId, acbsFacilityGuaranteeInput);
 
     if (isHttpErrorStatus(status)) {
       throw new Error(
         JSON.stringify(
           {
             name: 'ACBS Facility Guarantee amend error',
-            facilityIdentifier,
```

Comment:
how come this isn't included as facilityId in the error message anymore?

Reply:
It is in `data` object.